samples/other/file-transfer-s3-to-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/demo/vault/SingletonVaultExtension.java

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright (c) 2021 Siemens AG
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Siemens AG - initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.transfer.demo.vault;
+
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.security.CertificateResolver;
+import org.eclipse.dataspaceconnector.spi.security.PrivateKeyResolver;
+import org.eclipse.dataspaceconnector.spi.security.Vault;
+import org.eclipse.dataspaceconnector.spi.system.VaultExtension;
+
+public class SingletonVaultExtension implements VaultExtension {
+
+    private Vault vault;
+
+    @Override
+    public void initialize(Monitor monitor) {
+        vault = initializeVault();
+
+        monitor.info("Initialized Singleton Vault extension");
+    }
+
+    @Override
+    public Vault getVault() {
+        return vault;
+    }
+
+    @Override
+    public PrivateKeyResolver getPrivateKeyResolver() {
+        return null;
+    }
+
+    @Override
+    public CertificateResolver getCertificateResolver() {
+        return null;
+    }
+
+    private final String SECRET_PROPERTY = "creds";
```

Comment:
one stylistic nit: can you move this to the top of the class and make it static so it is consistent with other classes?

Reply:
about code formatting: please refer to the [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) and associated editor configs to keep code style consistent.

Reply:
I used the codestyle for the IntelliJ IDEA ([here](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/main/resources)). Code inspection with the CheckStyle plugin points to a deprecated API - Metadatastore, which is crucial for the demo at the moment.

Reply:
The MetadataStore will be replaced with AssetIndex. It's OK for the time being to use the deprecated class

The SECRET_PROPERY should be moved up to the top of the class and use a descriptive string with "." similar to the previous examples. Thanks!

Reply:
I believe that's been done already. It now looks like this (at the beginning of the class):

```
@EdcSetting
private static final String CREDS = "dataspaceconnector.transfer.demo.s3.destination.creds";
```

Reply:
@jimmarino Is it OK?

Reply:
Looks good. I'll squash and merge. As a heads up, we had to make a change to StatusChecker (#95) that was just merged as well. This change should not impact you, but if your end-to-end test does not complete or throw an error, it could be related to that change. If so, ping either myself or @beardyinc.